aiosmtpd/smtp.py

```diff
@@ -74,8 +73,16 @@ def __init__(self, handler,
         self.require_starttls = tls_context and require_starttls
         self._tls_handshake_failed = False
         self._tls_protocol = None
+        self.session = None  # type: SessionContext
+        self.message = None  # type: MessageContext
```

Comment:
I don't think we need the hanging comments.

Another question is whether we need an ABC or interface to describe this.  Probably not (we can live with duck typing), but we *do* need more documentation, and this would have to go there.